### PR TITLE
inherit clap and toml crates from workspace

### PR DESCRIPTION
**Pull request by PradeepKiruvale**
## Proposed changes

Inherit the `clap` and `toml` from workspace Cargo.toml.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
#2181

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: crates/core/tedge_agent/src/state_repository/state.rs [hunk 1]
```diff
@@ -28,7 +28,10 @@ impl StateRepository for AgentStateRepository {
 
     async fn load(&self) -> Result<State, StateError> {
         match fs::read(&self.state_repo_path).await {
-            Ok(bytes) => Ok(toml::from_slice::<State>(bytes.as_slice())?),
+            Ok(bytes) => {
+                dbg!(&String::from_utf8(bytes.clone())?);
```
Comment (didier-wenzek) [inline crates/core/tedge_agent/src/state_repository/state.rs:32]: `dbg!`  to be removed

File: crates/extensions/c8y_config_manager/Cargo.toml [hunk 2]
```diff
@@ -31,3 +31,4 @@ toml = { workspace = true }
 tedge_actors = { workspace = true, features = ["test-helpers"] }
 tedge_test_utils = { workspace = true }
 tokio = { workspace = true, features = ["time", "test-util"] }
+toml = "0.7"
```
Comment (didier-wenzek) [inline crates/extensions/c8y_config_manager/Cargo.toml:34]: Why not `{ workspace = true }` ?

File: crates/tests/tedge_test_utils/src/fs.rs [hunk 3]
```diff
@@ -88,7 +88,7 @@ impl TempTedgeFile {
         file.sync_all().unwrap();
     }
 
-    pub fn with_toml_content(self, content: toml::Value) {
+    pub fn with_toml_content(self, content: toml::Table) {
```
Comment (didier-wenzek) [inline crates/tests/tedge_test_utils/src/fs.rs:91]: What is the motivation for this change? The former version were more generic.

File: plugins/tedge_apt_plugin/Cargo.toml [hunk 4]
```diff
@@ -10,7 +10,7 @@ homepage = { workspace = true }
 repository = { workspace = true }
 
 [dependencies]
-clap = { version = "3", features = ["derive"] }
+clap = { workspace = true, features = ["derive"] }
```
Comment (didier-wenzek) [inline plugins/tedge_apt_plugin/Cargo.toml:13]: The system test errors are likely related to this change: https://github.com/thin-edge/thin-edge.io/actions/runs/6039139197/job/16387472405?pr=2206.

```
`sudo /etc/tedge/sm-plugins/apt...` returned an unexpected exit code

  error: unexpected argument '--name' found

  tip: a similar argument exists: '----name'

Usage: apt list <----name <NAME>|----maintainer <MAINTAINER>>
```

File: crates/extensions/c8y_config_manager/Cargo.toml (hunk 2 above)
Comment (PradeepKiruvale) [reply crates/extensions/c8y_config_manager/Cargo.toml:34]: Yes, it can be inherited. I fixed it.

File: crates/core/tedge_agent/src/state_repository/state.rs (hunk 1 above)
Comment (PradeepKiruvale) [reply crates/core/tedge_agent/src/state_repository/state.rs:32]: removed

File: crates/tests/tedge_test_utils/src/fs.rs (hunk 3 above)
Comment (PradeepKiruvale) [reply crates/tests/tedge_test_utils/src/fs.rs:91]: In the toml `0.7` version, the `toml` macro builds the `Table`. So, I had to update this.

File: plugins/tedge_apt_plugin/Cargo.toml (hunk 4 above)
Comment (PradeepKiruvale) [reply plugins/tedge_apt_plugin/Cargo.toml]: Yes, I fixed it.

**Review by didier-wenzek — APPROVED** [review]
Approved. Thank you.
